### PR TITLE
Decouple enabling ACL and auth for ZK connection

### PR DESCRIPTION
**Pull request by rohansingh**
If ZK credentials are specified, use authentication for the connection,
even if ACL's are not enabled.

### CODE REVIEW

**Comment by rohansingh** [comment]
@gimaker

File: helios-services/src/main/java/com/spotify/helios/agent/AgentService.java [hunk 1]
```diff
@@ -328,12 +328,17 @@ private ZooKeeperClient setupZookeeperClient(final AgentConfig config, final Str
...
-      final String agentUser = config.getZookeeperAclAgentUser();
-      final String agentPassword = config.getZooKeeperAclAgentPassword();
-      final String masterUser = config.getZookeeperAclMasterUser();
-      final String masterDigest = config.getZooKeeperAclMasterDigest();
+    final String agentUser = config.getZookeeperAclAgentUser();
+    final String agentPassword = config.getZooKeeperAclAgentPassword();
+    final String masterUser = config.getZookeeperAclMasterUser();
+    final String masterDigest = config.getZooKeeperAclMasterDigest();
+
+    if (!isNullOrEmpty(agentUser)) {
```
Comment (gimaker) [inline helios-services/src/main/java/com/spotify/helios/agent/AgentService.java]: check for empty/null agent password

**Comment by gimaker** [comment]
:+1:

**Comment by mattnworb** [comment]
makes a lot of sense :+1:

**Pull request merged**